docs/standard/analyzers/platform-compat-analyzer.md

```diff
@@ -0,0 +1,363 @@
+
+4. **Delete the code**. Generally not what you want because it means you lose fidelity when your code is used by Windows users, but for cases where a cross-platform alternative exists, you’re likely better off using that over platform-specific APIs.
+
+5. **Suppress the warning**. You can of course cheat and simply suppress the warning, either via editor.config or #pragma warning disable. However, you should prefer options (1), (2) and (3) when using platform-specific APIs.
+
+### The platform specific API calls in the above examples guarded with guard methods
+
+Guard methods platform name should match with the calling platform dependent API platform name. If platform string of the calling API include version:
+
+- For `[SupportedOSPlatform("platformVersion")]` attribute, guard method platform `version` should be >= calling platform `Version`
+- For `[UnsupportedOSPlatform("platformVersion")]` attribute, guard method platform `version` should be <= calling platform `Version`
+
+  ```csharp
+  public void CallingSupportedOnlyApis() // Allow list calls
+  {
+      if (OperatingSystem.IsWindows())
+      {
+          WindowsOnlyApi(); // not warn
+      }
+
+      if (OperatingSystem.IsLinux())
+      {
+          SupportedOnWindowsAndLinuxOnly(); // nor warn, within one of the supported context
+      }
+
+      if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows)) // traditional method
+      {
+          SupportedOnWindowsAndLinux(); // not warn
+      }
+
+      // Can use &&, || logical operators to guard combined attributes
+      if (OperatingSystem.IsWindowsVersionAtLeast(8) && !OperatingSystem.IsWindowsVersionAtLeast(10.0.19041)))
+      {
+          ApiSupportedFromWindows8UnsupportFromWindows10();
+      }
+
+      if (OperatingSystem.IsWindowsVersionAtLeast(10, 0, 1903))
+      {
+          AssemblySupportedOnWindowsApiSupportedFromWindows10(); // Only need to check latest supported version
+      }
+  }
+
+  public void CallingUnsupportedApis() // Deny list calls
+  {
+      if (!OperatingSystem.IsAndroid())
+      {
+          DoesNotWorkOnAndroid(); // no diagnostic
+      }
+
+      if (!OperatingSystem.IsWindows() || OperatingSystem.IsWindowsVersionAtLeast(8))
+      {
+          StartedWindowsSupportFromVersion8(); // no diagnostic
+      }
+
+      if (!OperatingSystem.IsWindows() || // supported all other platform
+         (OperatingSystem.IsWindowsVersionAtLeast(8) && !OperatingSystem.IsWindowsVersionAtLeast(10, 0, 1903)))
+      {
+          StartedWindowsSupportFrom8UnsupportedFrom10(); // no diagnostic
+      }
+  }
+  ```
+
+### Call site marked as platform specific for platform specific API calls in above examples
+
+Platform names should match with the calling platform dependent API. If platform string include version:
+
+- For `[SupportedOSPlatform("platformVersion")]` attribute, the call site platform `version` should be >= calling platform `Version`
```

Comment:
Please make same changes suggested in the previous section for these two bullet points.